src/backend/distributed/utils/ruleutils_11.c

```diff
@@ -7837,45 +7836,14 @@ generate_operator_name(Oid operid, Oid arg1, Oid arg2)
 	oprname = NameStr(operform->oprname);
 
 	/*
-	 * The idea here is to schema-qualify only if the parser would fail to
-	 * resolve the correct operator given the unqualified op name with the
-	 * specified argtypes.
+	 * Unlike generate_operator_name() in postgres/src/backend/utils/adt/ruleutils.c,
+	 * we don't check if the operator is in current namespace or not. This is
+	 * because this check is costly when the operator is not in current namespace.
 	 */
-	switch (operform->oprkind)
-	{
-		case 'b':
-			p_result = oper(NULL, list_make1(makeString(oprname)), arg1, arg2,
-							true, -1);
-			break;
-		case 'l':
-			p_result = left_oper(NULL, list_make1(makeString(oprname)), arg2,
-								 true, -1);
-			break;
-		case 'r':
-			p_result = right_oper(NULL, list_make1(makeString(oprname)), arg1,
-								  true, -1);
-			break;
-		default:
-			elog(ERROR, "unrecognized oprkind: %d", operform->oprkind);
-			p_result = NULL;	/* keep compiler quiet */
-			break;
-	}
-
-	if (p_result != NULL && oprid(p_result) == operid)
-		nspname = NULL;
-	else
-	{
-		nspname = get_namespace_name(operform->oprnamespace);
-		appendStringInfo(&buf, "OPERATOR(%s.", quote_identifier(nspname));
-	}
-
+	nspname = get_namespace_name(operform->oprnamespace);
```

Comment:
Is there any reason this might be `NULL`? If so, can you add an `Assert` here?

Reply:
Unless user messes with `pg_operator` manually, this shouldn't be `NULL`. I added an assert here. Postgres' `generate_operator_name()` assumes this cannot be `NULL`. They don't have an `Assert()` here, but the following `quote_identifier(nspname)` will crash if this is `NULL`, so I guess they assume this is always not null.